test/integration/api/entities.js

```diff
@@ -0,0 +1,263 @@
+            source: {
+              type: 'api',
+              details: null
+            },
+            label: 'Jane Roe',
+            versionNumber: 2
+          });
+          body[1].action.should.be.eql('entity.create');
+          // assert nested logs here
+        });
+    }));
+  });
+
+  describe('POST /datasets/:name/entities', () => {
+
+    it('should create an Entity', testService(async (service) => {
+      const asAlice = await service.login('alice');
+
+      await asAlice.post('/v1/projects/1/datasets/People/entities')
+        .send({
+          uuid: '10000000-0000-0000-0000-000000000001',
+          label: 'Johnny Doe',
+          firstName: 'Johnny',
+          lastName: 'Doe',
+          city: 'Toronto'
+        })
+        .expect(200)
+        .then(({ body: person }) => {
+          person.should.be.an.Entity();
+          person.should.have.property('currentVersion').which.is.an.EntityDef();
+          person.currentVersion.should.have.property('source').which.is.an.EntitySource();
+          person.currentVersion.should.have.property('data').which.is.eql({
+            firstName: 'Johnny',
+            lastName: 'Doe',
+            city: 'Toronto'
+          });
+        });
+    }));
+
+    // it should reject if uuid or label is missing
+    // it should reject if property is not present in dataset.publishedProperties
+    // it should reject if user don't have permission
+    // it should reject if uuid is not unique ??? what to do if uuid is deleted?
+
+  });
+
+  describe('PUT /datasets/:name/entities/:uuid', () => {
+
+    it('should update an Entity', testService(async (service) => {
+      const asAlice = await service.login('alice');
+
+      await asAlice.put('/v1/projects/1/datasets/People/entities/10000000-0000-0000-0000-000000000001')
+        .send({
+          uuid: '10000000-0000-0000-0000-000000000001',
+          label: 'Richard Roe',
+          firstName: 'Richard',
+          lastName: 'Roe',
+          city: 'Toronto'
+        })
+        .expect(200)
+        .then(({ body: person }) => {
+          person.should.be.an.Entity();
+          person.should.have.property('currentVersion').which.is.an.EntityDef();
+          person.currentVersion.should.have.property('source').which.is.an.EntitySource();
+          person.currentVersion.should.have.property('data').which.is.eql({
+            firstName: 'Richard',
+            lastName: 'Roe',
+            city: 'Toronto'
+          });
+        });
+    }));
+
+    // it should reject if uuid is not found
+    // it should reject if uuid in queryParam and body don't match
+    // it should reject if uuid or label is missing
+    // it should reject if property is not present in dataset.publishedProperties
+    // it should reject if user don't have permission
+  });
+
+  describe('PATCH /datasets/:name/entities/:uuid', () => {
+
+    it('should partially update an Entity', testService(async (service) => {
```

Comment:
I'm trying to understand the intended difference between PUT and PATCH here. Not that we have to figure it out now, though!

Reply:
PUT requires complete object. PATCH allows single field to be updated